### PR TITLE
Updated RTD for lambda executor

### PR DESCRIPTION
**Pull request by Emmanuel289**
- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

Closes https://github.com/AgnostiqHQ/covalent-staging/issues/601

### CODE REVIEW

File: doc/source/api/executors/awslambda.rst [hunk 1]
```diff
@@ -7,23 +7,193 @@
...
+     - Config Name
+     - Description
+   * - IAM Role
+     - lambda_role_name
+     - The IAM role this lambda will assume during execution of your tasks
+   * - S3 Bucket
+     - s3_bucket_name
+     - Name of an AWS S3 bucket that the executor can use to store temporary files
+
+4a. Manually provisioning resources
```
Comment (AlejandroEsquivel) [inline doc/source/api/executors/awslambda.rst:192]: @Emmanuel289 I think just due to aesthetic reasons let's just remove the `4a. Manually provisioning resources` heading (also such that it is consistent with the other executor's updated RTD)

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]
Hey @Emmanuel289 I think so far the content is great, just a few minor changes needed

File: doc/source/api/executors/awslambda.rst (hunk 1 above)
Comment (Emmanuel289) [reply doc/source/api/executors/awslambda.rst]: > @Emmanuel289 I think just due to aesthetic reasons let's just remove the `4a. Manually provisioning resources` heading (also such that it is consistent with the other executor's updated RTD)

I have effected the requested changes. Thanks!

File: doc/source/api/executors/awslambda.rst [hunk 2]
```diff
@@ -7,23 +7,193 @@
...
-uses the S3 and the AWS lambda service to execute tasks thus the IAM roles and policies must be configured so that
-the executor has permissions to interact with the two. The following JSON policy document shows the necessary IAM
-permissions required for the executor to properly run tasks using the AWS Lambda compute service
+2. Usage Example
+################
+
+This is an example of how a workflow can be constructed to use the AWS Lambda executor. In the example, we join two words to form a phrase
+and return an excited phrase.
+
+.. code:: python
```
[the remark this answers is not in the document]
Comment (Emmanuel289) [reply doc/source/api/executors/awslambda.rst]: Done

**Pull request merged**